render_list.cpp

```diff
@@ -59,13 +59,20 @@ static bool render_list(std::ostream & s, int32_t page, const std::string & pref
 {
     auto begin = get_first<T>();
 
+    // get page count
     int32_t page_count = (int32_t(get_vector<T>().end() - begin) + items_per_page - 1) / items_per_page;
+    
+    // return if invalid page number
     if (page < 0 || page >= page_count)
     {
         return false;
     }
 
-    s << "<!DOCTYPE html><html><head><meta charset=\"utf-8\"><title>" << title << ", page " << page << "</title><meta name=\"viewport\" content=\"width=device-width, initial-scale=1\"><link rel=\"stylesheet\" href=\"style.css\"></head><body><h1>" << title << "</h1><ul>";
+    s << "<!DOCTYPE html><html><head><meta charset=\"utf-8\"><title>" << title << ", page " << page << "</title><meta name=\"viewport\" content=\"width=device-width, initial-scale=1\">";
+    
+    add_styles_and_scripts(s);
```

Comment:
jQuery is quite large, so it should only be included on pages that actually use it.